### ISSUE TITLE
Out of bound indices will not be tracked by MobX

### ISSUE THREAD

**Issue by ScottSpittle** [500tech/angular-tree-component#390]
I still seem to be getting the issue reported here https://github.com/500tech/angular-tree-component/issues/195

`[mobx.array] Attempt to read an array index (0) that is out of bounds (0). Please check length first. Out of bound indices will not be tracked by MobX`

**Comment by johnking** [comment]
I have the same issue when I use `findNextNode()` as specified in  [#419](https://github.com/500tech/angular-tree-component/issues/419) 

```
mobx.js:571 [mobx.array] Attempt to read an array index (2) that is out of bounds (2). Please check length first. Out of bound indices will not be tracked by MobX
webpackJsonp.../../../../mobx/lib/mobx.js.ObservableArray.get @ mobx.js:571
get @ mobx.js:685
webpackJsonp.../../../../angular-tree-component/dist/models/tree-node.model.js.TreeNode._findAdjacentSibling @ tree-node.model.js:145
webpackJsonp.../../../../angular-tree-component/dist/models/tree-node.model.js.TreeNode.findNextSibling @ tree-node.model.js:149
webpackJsonp.../../../../angular-tree-component/dist/models/tree-node.model.js.TreeNode.findNextNode @ tree-node.model.js:179
webpackJsonp.../../../../../src/app/kepan/chapter-kepan/chapter-kepan.component.ts.ChapterKepanComponent.nextVerse @ chapter-kepan.component.ts:191
(anonymous) @ ChapterKepanComponent.html:15
handleEvent @ core.es5.js:12014
callWithDebugContext @ core.es5.js:13475
debugHandleEvent @ core.es5.js:13063
dispatchEvent @ core.es5.js:8607
(anonymous) @ core.es5.js:9218
(anonymous) @ platform-browser.es5.js:2651
webpackJsonp.../../../../zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:425
onInvokeTask @ core.es5.js:3881
webpackJsonp.../../../../zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:424
webpackJsonp.../../../../zone.js/dist/zone.js.Zone.runTask @ zone.js:192
webpackJsonp.../../../../zone.js/dist/zone.js.ZoneTask.invokeTask @ zone.js:499
invokeTask @ zone.js:1427
globalZoneAwareCallback @ zone.js:1445
mobx.js:571 [mobx.array] Attempt to read an array index (6) that is out of bounds (6). Please check length first. Out of bound indices will not be tracked by MobX
webpackJsonp.../../../../mobx/lib/mobx.js.ObservableArray.get @ mobx.js:571
get @ mobx.js:685
webpackJsonp.../../../../angular-tree-component/dist/models/tree-node.model.js.TreeNode._findAdjacentSibling @ tree-node.model.js:145
webpackJsonp.../../../../angular-tree-component/dist/models/tree-node.model.js.TreeNode.findNextSibling @ tree-node.model.js:149
webpackJsonp.../../../../angular-tree-component/dist/models/tree-node.model.js.TreeNode.findNextNode @ tree-node.model.js:179
webpackJsonp.../../../../../src/app/kepan/chapter-kepan/chapter-kepan.component.ts.ChapterKepanComponent.nextVerse @ chapter-kepan.component.ts:191
(anonymous) @ ChapterKepanComponent.html:15
handleEvent @ core.es5.js:12014
callWithDebugContext @ core.es5.js:13475
debugHandleEvent @ core.es5.js:13063
dispatchEvent @ core.es5.js:8607
(anonymous) @ core.es5.js:9218
(anonymous) @ platform-browser.es5.js:2651
webpackJsonp.../../../../zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:425
onInvokeTask @ core.es5.js:3881
webpackJsonp.../../../../zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:424
webpackJsonp.../../../../zone.js/dist/zone.js.Zone.runTask @ zone.js:192
webpackJsonp.../../../../zone.js/dist/zone.js.ZoneTask.invokeTask @ zone.js:499
invokeTask @ zone.js:1427
globalZoneAwareCallback @ zone.js:1445
mobx.js:571 [mobx.array] Attempt to read an array index (2) that is out of bounds (2). Please check length first. Out of bound indices will not be tracked by MobX
webpackJsonp.../../../../mobx/lib/mobx.js.ObservableArray.get @ mobx.js:571
get @ mobx.js:685
webpackJsonp.../../../../angular-tree-component/dist/models/tree-node.model.js.TreeNode._findAdjacentSibling @ tree-node.model.js:145
webpackJsonp.../../../../angular-tree-component/dist/models/tree-node.model.js.TreeNode.findNextSibling @ tree-node.model.js:149
webpackJsonp.../../../../angular-tree-component/dist/models/tree-node.model.js.TreeNode.findNextNode @ tree-node.model.js:179
webpackJsonp.../../../../angular-tree-component/dist/models/tree-node.model.js.TreeNode.findNextNode @ tree-node.model.js:180
webpackJsonp.../../../../../src/app/kepan/chapter-kepan/chapter-kepan.component.ts.ChapterKepanComponent.nextVerse @ chapter-kepan.component.ts:191
(anonymous) @ ChapterKepanComponent.html:15
handleEvent @ core.es5.js:12014
callWithDebugContext @ core.es5.js:13475
debugHandleEvent @ core.es5.js:13063
dispatchEvent @ core.es5.js:8607
(anonymous) @ core.es5.js:9218
(anonymous) @ platform-browser.es5.js:2651
webpackJsonp.../../../../zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:425
onInvokeTask @ core.es5.js:3881
webpackJsonp.../../../../zone.js/dist/zone.js.ZoneDelegate.invokeTask @ zone.js:424
webpackJsonp.../../../../zone.js/dist/zone.js.Zone.runTask @ zone.js:192
webpackJsonp.../../../../zone.js/dist/zone.js.ZoneTask.invokeTask @ zone.js:499
invokeTask @ zone.js:1427
globalZoneAwareCallback @ zone.js:1445
mobx.js:571 [mobx.array] Attempt to read an array index (2) that is out of bounds (2). Please check length first. Out of bound indices will not be tracked by MobX
```

**Comment by benthedeveloper** [comment]
Same warning appears if you try to use findNextSibling(), and there is no next sibling node.
One workaround might be to check if (node === node.parent.getLastChild()) to tell if it is the last node before calling findNext(false) or findNextSibling(), assuming the node has a parent. I'd rather the warning not appear though, as the documentation says it should just return null if next sibling doesn't exist.

**Comment by adamkleingit** [comment]
Fixed in 5.2.1 - can you verify?
